### PR TITLE
add bwc test for reports-scheduler

### LINKED ISSUE #240: Add backwards compatibility tests for automation
Add backwards compatibility tests for the plugin to automate end-to-end upgrade paths for faster releases. Developer documentation for implementing bwc tests and hooking them to CI: https://github.com/opensearch-project/opensearch-plugins/blob/main/TESTING.md#backwards-compatibility-testing

### PR DESCRIPTION
**Pull request by zhongnansu**
Signed-off-by: Zhongnan Su <szhongna@amazon.com>

### Description
- Add bwc specific IT
- update gradle.build to run bwc tests from opendistro 1.13 to current version(1.3-snapshot)
- update CI workflow to include bwc test

### TODO
- need to improve bwc testing  from version to version, e.g(7.10-> 1.1, 7.10 -> 1.2, 1.1 -> 1.2, 1.2 -> 1.3)

### Issues Resolved
#240 

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: reports-scheduler/build.gradle [hunk 1]
```diff
@@ -255,8 +268,180 @@ testClusters.integTest {
...
+                }
+            }
+        }),
+        provider(new Callable<RegularFile>(){
+            @Override
+            RegularFile call() throws Exception {
+                return new RegularFile() {
+                    @Override
+                    File getAsFile() {
+                        return fileTree(bwcFilePath + "/reports-scheduler/" + project.version).getSingleFile()
```
Comment (joshuali925) [inline reports-scheduler/build.gradle:338]: there should be a way to not include the `1.3.0.0-SNAPSHOT` like other bwc PRs.

SQL is actually having the same issue, but SQL somehow doesn't have `bundlePlugin`. what alerting did might work here https://github.com/opensearch-project/alerting/blob/main/alerting/build.gradle#L204-L214

**Review by joshuali925 — DISMISSED** [review]

File: .github/workflows/dashboards-reports-test-and-build-workflow.yml [hunk 2]
```diff
@@ -32,6 +32,7 @@ jobs:
 
       - name: Add Chromium Binary to Reporting for Testing
         run: |
+          sudo apt update
```
Comment (davidcui1225) [inline .github/workflows/dashboards-reports-test-and-build-workflow.yml:35]: I think this change is covered in Josh's PR  #245 , can you merge from upstream here?

File: reports-scheduler/build.gradle (hunk 1 above)
Comment (zhongnansu) [reply reports-scheduler/build.gradle]: I was following AD and they added that. But I see what you mean, maybe we can get rid of 1.3.0-snapshot for reports-scheduler, but we'll need the job-scheduler. I'll run some test

**Review by davidcui1225 — APPROVED** [review]

**Review by kavithacm — APPROVED** [review]

File: reports-scheduler/build.gradle (hunk 1 above)
Comment (zhongnansu) [reply reports-scheduler/build.gradle]: changed to remove the reports-scheduler latest artifacts in `src/test/resources`

**Review by joshuali925 — APPROVED** [review]

**Pull request merged**